### ISSUE TITLE
Error: Could not find or load main class com.ethercis.dao.access.support.TerminologySetter

### ISSUE THREAD

**Issue by daniloblb** [skoba/docker-ethercis#1]
Attaching to docker-ethercis_postgres_1, docker-ethercis_app_1
postgres_1  | 2018-10-17 18:31:16.568 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1  | 2018-10-17 18:31:16.568 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1  | 2018-10-17 18:31:16.600 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1  | 2018-10-17 18:31:16.890 UTC [21] LOG:  database system was shut down at 2018-10-17 18:15:49 UTC
postgres_1  | 2018-10-17 18:31:17.134 UTC [1] LOG:  database system is ready to accept connections
postgres_1  | 2018-10-17 18:31:18.209 UTC [28] LOG:  incomplete startup packet
postgres_1  | 2018-10-17 18:31:26.007 UTC [29] ERROR:  relation "ehr.template" does not exist at character 210
postgres_1  | 2018-10-17 18:31:26.007 UTC [29] STATEMENT:  select "ehr"."template"."uid", "ehr"."template"."template_id", "ehr"."template"."concept", "ehr"."template"."introspect", "ehr"."template"."parsed_opt", "ehr"."template"."visitor", "ehr"."template"."crc" from "ehr"."template" where "ehr"."template"."introspect" is not null
postgres_1  | 2018-10-17 19:01:22.391 UTC [1] LOG:  received smart shutdown request
postgres_1  | 2018-10-17 19:01:22.439 UTC [1] LOG:  worker process: logical replication launcher (PID 27) exited with exit code 1
postgres_1  | 2018-10-17 19:01:32.413 UTC [93] FATAL:  terminating connection due to unexpected postmaster exit
postgres_1  | 2018-10-17 19:01:32.415 UTC [39] FATAL:  terminating connection due to unexpected postmaster exit
postgres_1  | 2018-10-17 19:01:58.161 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1  | 2018-10-17 19:01:58.161 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1  | 2018-10-17 19:01:58.209 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1  | 2018-10-17 19:01:58.320 UTC [20] LOG:  database system was interrupted; last known up at 2018-10-17 19:01:22 UTC
postgres_1  | 2018-10-17 19:01:58.565 UTC [20] LOG:  database system was not properly shut down; automatic recovery in progress
postgres_1  | 2018-10-17 19:01:58.590 UTC [20] LOG:  redo starts at 0/18C6978
postgres_1  | 2018-10-17 19:01:58.590 UTC [20] LOG:  invalid record length at 0/18C6A20: wanted 24, got 0
postgres_1  | 2018-10-17 19:01:58.590 UTC [20] LOG:  redo done at 0/18C69B0
postgres_1  | 2018-10-17 19:01:58.704 UTC [1] LOG:  database system is ready to accept connections
postgres_1  | 2018-10-17 19:01:58.912 UTC [27] LOG:  incomplete startup packet
postgres_1  | 2018-10-17 19:02:04.588 UTC [28] ERROR:  column template.introspect does not exist at character 92
postgres_1  | 2018-10-17 19:02:04.588 UTC [28] STATEMENT:  select "ehr"."template"."uid", "ehr"."template"."template_id", "ehr"."template"."concept", "ehr"."template"."introspect", "ehr"."template"."parsed_opt", "ehr"."template"."visitor", "ehr"."template"."crc" from "ehr"."template" where "ehr"."template"."introspect" is not null
app_1       | wait-for-it.sh: waiting 60 seconds for postgres:5432
postgres_1  | 2018-10-18 17:16:03.198 UTC [2685] LOG:  incomplete startup packet
app_1       | wait-for-it.sh: postgres:5432 is available after 0 seconds
app_1       | populating concept table
app_1       | Error: Could not find or load main class com.ethercis.dao.access.support.TerminologySetter
app_1       | ethercis startup
app_1       | tail: unrecognized file system type 0x794c7630 for ‘/var/opt/ecis/ethercis_test.log’. please report this to bug-coreutils@gnu.org. reverting to polling
app_1       | Listening for transport dt_socket at address: 8000
app_1       | ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console.
app_1       | Property: File 'services.properties' not found
app_1       | Property: could not find default property file, will try from command line arguments
app_1       | Property: No args array given.
app_1       | Property: Loading /etc/opt/ecis/services.properties from directory /home/ethercis
app_1       | Property: Added 3 property pairs
app_1       | SLF4J: Class path contains multiple SLF4J bindings.
app_1       | SLF4J: Found binding in [jar:file:/opt/ecis/lib/deploy/ecis-ehrdao-1.2.0-SNAPSHOT.jar!/org/slf4j/impl/StaticLoggerBinder.class]
app_1       | SLF4J: Found binding in [jar:file:/opt/ecis/lib/deploy/ecis-meta-data-cache-1.2.0-SNAPSHOT.jar!/org/slf4j/impl/StaticLoggerBinder.class]
app_1       | SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
app_1       | SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
app_1       | [main] INFO org.eclipse.jetty.util.log - Logging initialized @1829ms to org.eclipse.jetty.util.log.Slf4jLog
app_1       | Property: Loading /etc/opt/ecis/services.properties from directory /home/ethercis
app_1       | Property: Added 3 property pairs
app_1       | 17:16:06.702 [main] ERROR com.ethercis.servicemanager.jmx.JmxWrapper - JMX registration problem for 'com.ethercis:nodeClass=node,node="logging"': javax.management.NotCompliantMBeanException: Bad getMBeanInfo()
app_1       | javax.management.NotCompliantMBeanException: Bad getMBeanInfo()
app_1       |   at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getNewMBeanClassName(DefaultMBeanServerInterceptor.java:336)
app_1       |   at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerMBean(DefaultMBeanServerInterceptor.java:319)
app_1       |   at com.sun.jmx.mbeanserver.JmxMBeanServer.registerMBean(JmxMBeanServer.java:522)
app_1       |   at com.ethercis.servicemanager.jmx.JmxWrapper.registerMBean(JmxWrapper.java:693)
app_1       |   at com.ethercis.servicemanager.jmx.JmxWrapper.registerMBean(JmxWrapper.java:673)
app_1       |   at com.ethercis.servicemanager.jmx.JmxWrapper.<init>(JmxWrapper.java:177)
app_1       |   at com.ethercis.servicemanager.jmx.JmxWrapper.getInstance(JmxWrapper.java:123)
app_1       |   at com.ethercis.servicemanager.cluster.RunTimeSingleton.getJmxWrapper(RunTimeSingleton.java:255)
app_1       |   at com.ethercis.servicemanager.cluster.RunTimeSingleton.registerMBean(RunTimeSingleton.java:304)
app_1       |   at com.ethercis.servicemanager.runlevel.RunlevelManager.initJmx(RunlevelManager.java:123)
app_1       |   at com.ethercis.servicemanager.cluster.RunTimeSingleton.getRunlevelManager(RunTimeSingleton.java:1614)
app_1       |   at com.ethercis.vehr.VEhrGateServlet.init(VEhrGateServlet.java:235)
app_1       |   at com.ethercis.vehr.Launcher.start(Launcher.java:239)
app_1       |   at com.ethercis.vehr.Launcher.startServer(Launcher.java:116)
app_1       |   at com.ethercis.vehr.Launcher.main(Launcher.java:271)
app_1       | Caused by: java.lang.NullPointerException
app_1       |   at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getNewMBeanClassName(DefaultMBeanServerInterceptor.java:333)
app_1       |   ... 14 more
postgres_1  | 2018-10-18 17:16:10.878 UTC [2686] ERROR:  relation "ehr.template" does not exist at character 210
postgres_1  | 2018-10-18 17:16:10.878 UTC [2686] STATEMENT:  select "ehr"."template"."uid", "ehr"."template"."template_id", "ehr"."template"."concept", "ehr"."template"."introspect", "ehr"."template"."parsed_opt", "ehr"."template"."visitor", "ehr"."template"."crc" from "ehr"."template" where "ehr"."template"."introspect" is not null
app_1       | [main] INFO org.jooq.Constants -
app_1       |
app_1       | @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
app_1       | @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
app_1       | @@@@@@@@@@@@@@@@  @@        @@@@@@@@@@
app_1       | @@@@@@@@@@@@@@@@@@@@        @@@@@@@@@@
app_1       | @@@@@@@@@@@@@@@@  @@  @@    @@@@@@@@@@
app_1       | @@@@@@@@@@  @@@@  @@  @@    @@@@@@@@@@
app_1       | @@@@@@@@@@        @@        @@@@@@@@@@
app_1       | @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
app_1       | @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
app_1       | @@@@@@@@@@        @@        @@@@@@@@@@
app_1       | @@@@@@@@@@    @@  @@  @@@@  @@@@@@@@@@
app_1       | @@@@@@@@@@    @@  @@  @@@@  @@@@@@@@@@
app_1       | @@@@@@@@@@        @@  @  @  @@@@@@@@@@
app_1       | @@@@@@@@@@        @@        @@@@@@@@@@
app_1       | @@@@@@@@@@@@@@@@@@@@@@@  @@@@@@@@@@@@@
app_1       | @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
app_1       | @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@  Thank you for using jOOQ 3.10.0
app_1       |
app_1       | errorCode=resource.configuration message=#1.0.0 Unable to setup DB layer accessorg.jooq.exception.DataAccessException: SQL [select "ehr"."template"."uid", "ehr"."template"."template_id", "ehr"."template"."concept", "ehr"."template"."introspect", "ehr"."template"."parsed_opt", "ehr"."template"."visitor", "ehr"."template"."crc" from "ehr"."template" where "ehr"."template"."introspect" is not null]; ERROR: relation "ehr.template" does not exist
app_1       |   Position: 210
app_1       |   at com.ethercis.persistence.ResourceService.doInit(ResourceService.java:99)
app_1       |   at com.ethercis.servicemanager.cluster.ClusterInfo.init(ClusterInfo.java:347)
app_1       |   at com.ethercis.servicemanager.service.ServiceManagerBase.instantiateServiceSecondPhase(ServiceManagerBase.java:293)
app_1       |   at com.ethercis.servicemanager.service.ServiceManagerBase.getServiceObject(ServiceManagerBase.java:126)
app_1       |   at com.ethercis.servicemanager.runlevel.RunlevelManager.startupServices(RunlevelManager.java:360)
app_1       |   at com.ethercis.servicemanager.runlevel.RunlevelManager.changeRunlevel(RunlevelManager.java:278)
app_1       |   at com.ethercis.vehr.VEhrGateServlet.init(VEhrGateServlet.java:238)
app_1       |   at com.ethercis.vehr.Launcher.start(Launcher.java:239)
app_1       |   at com.ethercis.vehr.Launcher.startServer(Launcher.java:116)
app_1       |   at com.ethercis.vehr.Launcher.main(Launcher.java:271)
app_1       | 17:16:10.904 [main] ERROR com.ethercis.vehr.VEhrGateServlet - ServiceManagerException errorCode=[resource.configuration.serviceFailed] location=[RunlevelManager.startupServices] message=[#1.0.0 Can't load plugin 'ResourceService']
app_1       | 17:16:10.905 [main] ERROR com.ethercis.vehr.VEhrGateServlet - Changing runlevel to 'RUNNING' failed, good bye
app_1       | Exception in thread "Signal catcher thread for controlled shudown" java.lang.NullPointerException
app_1       |   at com.ethercis.vehr.VEhrGateServlet.destroy(VEhrGateServlet.java:287)
app_1       |   at com.ethercis.vehr.VEhrGateServlet.shutdownHook(VEhrGateServlet.java:439)
app_1       |   at com.ethercis.servicemanager.cluster.SignalCatcher.run(SignalCatcher.java:216)
app_1       |   at java.lang.Thread.run(Thread.java:745)

**Comment by skoba** [comment]
This log looks fatal in PostgreSQL deployment.
Could you do "docker-compose down" and "docker-compose up" again?

**Comment by skoba** [comment]
I cannot make it clear by this log. Could you show me your configuration? OS, docker version, docker installation or other.

**Comment by daniloblb** [comment]
Shure.

 Ubuntu 18.04.1 LTS (GNU/Linux 4.15.0-36-generic x86_64)


root@ethercis-ubuntu:~# docker version
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:24:51 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:23:15 2018
  OS/Arch:          linux/amd64
  Experimental:     false

**Comment by daniloblb** [comment]
Thank you for reply.
I speak from Brazil, I am implementing the OpenEHR platform using EhterCIS, and I have some difficulties to increase the service and consume.
With regard to the error of postgres, I managed to solve it, I made some changes to the pgsql_ehr.ddl file.
I still have the following error in the application service:
app_1       | wait-for-it.sh: postgres:5432 is available after 8 seconds
app_1       | populating concept table
app_1       | Error: Could not find or load main class com.ethercis.dao.access.support.TerminologySetter
app_1       | ethercis startup
app_1       | tail: unrecognized file system type 0x794c7630 for ‘/var/opt/ecis/ethercis_test.log’. please report this to bug-coreutils@gnu.org. reverting to polling
app_1       | Listening for transport dt_socket at address: 8000
app_1       | ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console.
app_1       | Property: File 'services.properties' not found
app_1       | Property: could not find default property file, will try from command line arguments
app_1       | Property: No args array given.
app_1       | Property: Loading /etc/opt/ecis/services.properties from directory /home/ethercis
app_1       | Property: Added 3 property pairs
app_1       | SLF4J: Class path contains multiple SLF4J bindings.
app_1       | SLF4J: Found binding in [jar:file:/opt/ecis/lib/deploy/ecis-ehrdao-1.2.0-SNAPSHOT.jar!/org/slf4j/impl/StaticLoggerBinder.class]
app_1       | SLF4J: Found binding in [jar:file:/opt/ecis/lib/deploy/ecis-meta-data-cache-1.2.0-SNAPSHOT.jar!/org/slf4j/impl/StaticLoggerBinder.class]
app_1       | SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
app_1       | SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
app_1       | [main] INFO org.eclipse.jetty.util.log - Logging initialized @1992ms to org.eclipse.jetty.util.log.Slf4jLog
app_1       | Property: Loading /etc/opt/ecis/services.properties from directory /home/ethercis
app_1       | Property: Added 3 property pairs
app_1       | 17:20:27.446 [main] ERROR com.ethercis.servicemanager.jmx.JmxWrapper - JMX registration problem for 'com.ethercis:nodeClass=node,node="logging"': javax.management.NotCompliantMBeanException: Bad getMBeanInfo()
app_1       | javax.management.NotCompliantMBeanException: Bad getMBeanInfo()
app_1       |   at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getNewMBeanClassName(DefaultMBeanServerInterceptor.java:336)
app_1       |   at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerMBean(DefaultMBeanServerInterceptor.java:319)
app_1       |   at com.sun.jmx.mbeanserver.JmxMBeanServer.registerMBean(JmxMBeanServer.java:522)
app_1       |   at com.ethercis.servicemanager.jmx.JmxWrapper.registerMBean(JmxWrapper.java:693)
app_1       |   at com.ethercis.servicemanager.jmx.JmxWrapper.registerMBean(JmxWrapper.java:673)
app_1       |   at com.ethercis.servicemanager.jmx.JmxWrapper.<init>(JmxWrapper.java:177)
app_1       |   at com.ethercis.servicemanager.jmx.JmxWrapper.getInstance(JmxWrapper.java:123)
app_1       |   at com.ethercis.servicemanager.cluster.RunTimeSingleton.getJmxWrapper(RunTimeSingleton.java:255)
app_1       |   at com.ethercis.servicemanager.cluster.RunTimeSingleton.registerMBean(RunTimeSingleton.java:304)
app_1       |   at com.ethercis.servicemanager.runlevel.RunlevelManager.initJmx(RunlevelManager.java:123)
app_1       |   at com.ethercis.servicemanager.cluster.RunTimeSingleton.getRunlevelManager(RunTimeSingleton.java:1614)
app_1       |   at com.ethercis.vehr.VEhrGateServlet.init(VEhrGateServlet.java:235)
app_1       |   at com.ethercis.vehr.Launcher.start(Launcher.java:239)
app_1       |   at com.ethercis.vehr.Launcher.startServer(Launcher.java:116)
app_1       |   at com.ethercis.vehr.Launcher.main(Launcher.java:271)
app_1       | Caused by: java.lang.NullPointerException
app_1       |   at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getNewMBeanClassName(DefaultMBeanServerInterceptor.java:333)
app_1       |   ... 14 more
app_1       | [main] INFO org.jooq.Constants -
app_1       |

app_1       |
app_1       | [main] INFO org.eclipse.jetty.server.Server - jetty-9.4.z-SNAPSHOT; built: 2018-05-03T15:56:21.710Z; git: daa59876e6f384329b122929e70a80934569428c; jvm 1.8.0_111-8u111-b14-2~bpo8+1-b14
app_1       | Property: Added 0 property pairs
app_1       | [main] INFO org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@359ceb13{/,null,AVAILABLE}
app_1       | [main] INFO org.eclipse.jetty.server.AbstractConnector - Started ServerConnector@55cca199{HTTP/1.1,[http/1.1]}{4dc5fa6d2f6a:8080}
app_1       | [main] INFO org.eclipse.jetty.server.Server - Started @7725ms


### **As you can see the service went up, but with error.
I do not know what that can impact.
Can you help me ?**

**Comment by skoba** [comment]
OK. The environment seems enough to run this docker-image. But, I cannot follow your change from this code sets. Does this docker-image run in your environment without any change after git cloning?

**Comment by daniloblb** [comment]
Exactly, however in the table creation file, I put a CREATE for the template table, and in the docker compose file, I placed instead of version 10.4 for postgres I put the version 10, and then version 10.5 was installed.

**Comment by daniloblb** [comment]
Simply cloning the repository and executing the commands according to the documentation you have done, does not work. At least in my environment did not work. I would like to make this docker work and in parallel I am also trying to compile the code and make it work in another environment the service according to the documentation of Christian Chevalley that is in Git, but I have not yet been able to, in a second moment if you can help me too that would be great.

**Comment by daniloblb** [comment]
Although the service is running, in the boot log of the docker there are these three errors as I had already placed in the previous post.
Here are the three current mistakes:

1)
Error: Could not find or load main class com.ethercis.dao.access.support.TerminologySetter

2)
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console.
 Property: File 'services.properties' not found
 Property: could not find default property file, will try from command line arguments

3)
 ERROR com.ethercis.servicemanager.jmx.JmxWrapper - JMX registration problem for 'com.ethercis:nodeClass=node,node="logging"': javax.management.NotCompliantMBeanException: Bad getMBeanInfo()

**Comment by skoba** [comment]
I confirmed that the procedure shown bellow worked well.

 Clone the repository.
 Open a terminal in the directory where you cloned the repo.
 Run docker-compose up.

Did you installed OpenEMPI, too?

**Comment by daniloblb** [comment]
Without the changes I made it does not work in my environment. My hope was that you had already made the same mistake and could help me.

However, when docker docker which modules did you compile to provide the service?

According to the official documentation I have not been able to succeed yet.

Regarding openEMPI, I'm using yes, but I did not see docker.

In your etherCIS docker I still continue with the following error:

Error: Could not find or load main class com.ethercis.dao.access.support.TerminologySetter
